stdlib/public/Platform/bionic.modulemap.gyb

```diff
@@ -82,6 +82,10 @@ module SwiftGlibc [system] {
       header "${GLIBC_INCLUDE_PATH}/locale.h"
       export *
     }
+    module malloc {
+      header "${GLIBC_INCLUDE_PATH}/malloc.h"
+      export *
+    }
```

Comment:
`malloc.h` is deprecated and should not be included. It is supposed to be included from `stdlib.h`. This is a Linux-ism. That direct include from CoreFoundation can probably be replaced for `stdlib.h` and the problem will go away.

Reply:
OK, will try it.

Reply:
Tried the following patch, makes no difference:
```
diff --git a/CoreFoundation/Base.subproj/CoreFoundation_Prefix.h b/CoreFoundation/Base.subproj/CoreFoundation_Prefix.h
index 35e08036..4c06088e 100644
--- a/CoreFoundation/Base.subproj/CoreFoundation_Prefix.h
+++ b/CoreFoundation/Base.subproj/CoreFoundation_Prefix.h
@@ -284,7 +284,7 @@ typedef unsigned long fd_mask;
 #undef interface
 #endif
 
-#include <malloc.h>
+//#include <malloc.h>
 CF_INLINE size_t malloc_size(void *memblock) {
     return malloc_usable_size(memblock);
 }
diff --git a/CoreFoundation/Base.subproj/ForFoundationOnly.h b/CoreFoundation/Base.subproj/ForFoundationOnly.h
index 99407b78..fb469e55 100644
--- a/CoreFoundation/Base.subproj/ForFoundationOnly.h
+++ b/CoreFoundation/Base.subproj/ForFoundationOnly.h
@@ -51,7 +51,7 @@ CF_IMPLICIT_BRIDGING_DISABLED
 #pragma mark - CFRuntime
 
 #if TARGET_OS_LINUX
-#include <malloc.h>
+#include <stdlib.h>
 #elif TARGET_OS_BSD
 #include <stdlib.h> // malloc()
 #elif TARGET_OS_MAC
```
The error seems to indicate that it doesn't matter where the declaration comes from in the CoreFoundation headers, it cannot use that declaration in an `@inlinable` Swift function because CoreFoundation is implementation-only:
```
swift-corelibs-foundation/Sources/Foundation/Data.swift:88:20: error: global function 'malloc' cannot be used in an '@inlinable' function because 'CoreFoundation' was imported implementation-only
return malloc(size)
```
Rather, I think it was getting `malloc()` from the `Glibc` module that referenced prior NDKs, but something changed in the NDK sysroot from 22 onwards so `Glibc` couldn't find that C declaration anymore. Back then, I tried building with the _exact same_ source and Swift toolchain, but simply passing in the newer NDK and its clang to `build-script` and I'd get this error, implying an NDK sysroot change was interacting badly with our Swift import configuration somehow. 

I found this week that this header addition to the Bionic modulemap works around that but I don't know what causes this problem, unlike the libdispatch issue we discussed.

Reply:
I removed this header from this pull, as we only need it to compile the corelibs, so I'll add it to #38441 instead.